doc/Learn/Navigation/HowTo-DefineRoutes.md

```diff
@@ -0,0 +1,169 @@
+
+### Understanding routes
+
+In a new app with Navigation, you'll find some pre-set views: `Shell`, `MainPage`, and `SecondPage`. `Shell` acts as the main frame where views are set and navigation begins. When the app launches, it opens `MainPage`, which then leads to `SecondPage`.
+
+So if we take a look at the `RegisterRoutes` method in the `App.xaml.cs` file, we can see how these pages are organized in terms of routes:
+
+```csharp
+routes.Register(
+    new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+        Nested:
+        [
+            new ("Main", View: views.FindByViewModel<MainViewModel>()),
+            new ("Second", View: views.FindByViewModel<SecondViewModel>()),
+        ]
+    )
+);
+```
+
+We can observe that "Main" and "Second" are nested pages of "Shell", and both are at the same level. So if we take a look at the flow of the app, `ShellViewModel` opens `MainPage`, which in turn contains a button that navigates to `SecondPage`.
+
+#### RouteMap
+
+When defining routes, we utilize the `RouteMap` object. This object's constructor takes six arguments. Let's explore each of these arguments and how to effectively implement them:
+
+1. The first parameter is **Path**, serving as the identifier for the route. When navigating within your app using routes, this is the reference you'll use to reach the corresponding view.
+
+1. **View**: This parameter refers to the view associated with the route. The view must be registered beforehand within the `IViewRegistry`, and you can establish the association using the `FindByViewModel` method. Note in the following example how we add a route for a new page called `LoginPage`:
+
+    ```csharp
+    protected override void RegisterRoutes(IViewRegistry views, IRouteRegistry routes)
+    {
+        views.Register(
+            new ViewMap(ViewModel: typeof(ShellViewModel)),
+            new ViewMap<LoginPage, LoginViewModel>()
+        );
+
+        routes.Register(
+            new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+                Nested:
+                [
+                    new ("Login", View: views.FindByViewModel<LoginViewModel>())
+                ]
+            )
+        );
+    }
+    ```
+
+    Note that adding a view is not mandatory. You can specify a route without a view. For example, if you want to add routes for the content of a `TabBar`, you can define the routes without the view, and in your markup file, you can use the attached property `uen:Region.Name` to bind the content view with the defined route.
+
+    Given the following routes:
+
+    ```csharp
+    // Omitted for brevity
+    new RouteMap("Main", View: views.FindByViewModel<MainViewModel>(),
+        Nested:
+        [
+            new ("ForYouTab"),
+            new ("FavoritesTab")
+        ]
+    )
+    ```
+
+    We can specify the content view using the `uen:Region.Name` to link it to the given route name:
+
+    ```xml
+    <utu:TabBar uen:Region.Attached="True">
```

Comment:
This needs a parent Grid with Region.Attached="true" in order for it to work

Reply:
Fixed